src/components/AssignComponentMenu.vue

```
@@ -26,10 +38,37 @@ export default {
     };
   },
   methods: {
+    createComponentDefinition() {
+      this.$store.dispatch('models/createObjectWithType', { type: this.type })
+    },
+    deleteComponentDef(componentId) {
+      switch (this.type) {
+        case 'window_definitions':
+          this.$store.dispatch('models/destroyWindowDef', { object: { id: componentId } });
+          break;
+        case 'daylighting_control_definitions':
+          this.$store.dispatch('models/destroyDaylightingControlDef', { object: { id: componentId } });
+          break;
+        default:
+          this.$store.dispatch('models/destroyObject', { object: { id: componentId } });
```

let's throw an error here